plotly/tools.py

```diff
@@ -1508,21 +1507,14 @@ def _map_z2color(zval, colormap, vmin, vmax):
                                          "of vmax.")
         # find distance t of zval from vmin to vmax where the distance
         # is normalized to be between 0 and 1
-        t = (zval - vmin)/float((vmax - vmin))
-        t_color = FigureFactory._find_intermediate_color(colormap[0],
-                                                         colormap[1],
-                                                         t)
-        t_color = (t_color[0]*255.0, t_color[1]*255.0, t_color[2]*255.0)
-        labelled_color = 'rgb{}'.format(t_color)
-
-        return labelled_color
-
-    @staticmethod
-    def _tri_indices(simplices):
-        """
-        Returns a triplet of lists containing simplex coordinates
-        """
-        return ([triplet[c] for triplet in simplices] for c in range(3))
+        t = (zvals - vmin) / float((vmax - vmin))
+        t_colors = FigureFactory._find_intermediate_color(colormap[0],
+                                                          colormap[1],
+                                                          t)
+        t_colors = t_colors * 255.
+        labelled_colors = ['rgb(%s, %s, %s)' % (i, j, k)
```

Comment:
random speed thing I noticed - it turns out that python is a fair bit faster in generating strings with the % operator as opposed to `.format`, check out [this](http://tomatohater.com/2013/08/30/python-performance-string-formatting/).